### PR TITLE
Spike data cleansing in flink

### PR DESCRIPTION
**Pull request by ma-ro**
This PR implements
-DataProvider for default  Java data types
-DataCleansing operator for default Java data types
-Tests for DataCleansing operator

### CODE REVIEW

File: core/src/main/java/org/bptlab/cepta/operators/DataCleansingFunction.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/*This class implements the cleanseStream function which cleanses a DataSteam from one specific value.
+Currently supported are the following DataTypes:
+-Long
+-Float
+-Integer
+-String
+-Boolean
+*/
+public class DataCleansingFunction {
```
Comment (Quacck) [inline core/src/main/java/org/bptlab/cepta/operators/DataCleansingFunction.java]: Maybe we could use generics? :)

File: core/src/main/java/org/bptlab/cepta/operators/DataCleansingFunction.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      DataStream<Double> ResultStream = InputStream.filter(new FilterFunction<Double>() {
+          @Override
+          public boolean filter(Double value) throws Exception {
+            return (!(value.equals(FilterDouble)));
+          }
+        });
+      return ResultStream;
+  }
+    
+    public static DataStream<Integer> cleanseStream(DataStream<Integer> InputStream, Integer FilterInteger) {
```
Comment (Quacck) [inline core/src/main/java/org/bptlab/cepta/operators/DataCleansingFunction.java]: I think this wouldn't work well on our complex data types e.g. trainevents.

File: core/src/test/java/org/bptlab/cepta/data-cleansing-tests/DataCleansingTests.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.Ignore;
+import org.apache.flink.streaming.api.datastream.DataStreamUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.bptlab.cepta.operators.DataCleansingFunction;
+import org.bptlab.cepta.providers.JavaDataProvider;
+import sun.security.util.Length;
+
+public class DataCleansingTests {
```
Comment (Quacck) [inline core/src/test/java/org/bptlab/cepta/data-cleansing-tests/DataCleansingTests.java:18]: Please also test having a not-to-be-cleansed-value not getting cleansed

**Review by Quacck — CHANGES_REQUESTED** [review]
yoink

File: core/src/main/java/org/bptlab/cepta/operators/DataCleansingFunction.java (hunk 1 above)
Comment (ma-ro) [reply core/src/main/java/org/bptlab/cepta/operators/DataCleansingFunction.java]: we do now!

**Pull request merged**